File: src/test/java/LogoutTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class LogoutTest {

    @Test
    public void testLogoutFunctionality() {
        // Given the user is logged in
        simulateUserLogin(); // Simulate successful login

        // When the user clicks the 'Logout' button
        clickLogoutButton(); // Simulate clicking the logout button

        // Then check that the user is redirected to the login page
        assertEquals("loginPage", getCurrentPage());

        // And verify a message 'Logged out successfully' is displayed.
        assertEquals("Logged out successfully", getDisplayedMessage());
    }

    private void simulateUserLogin() {
        // Logic to simulate user login
    }

    private void clickLogoutButton() {
        // Logic to simulate clicking the logout button
    }

    private String getCurrentPage() {
        // Logic to get the current page
        return "loginPage"; // Placeholder return value
    }

    private String getDisplayedMessage() {
        // Logic to get the displayed message
        return "Logged out successfully"; // Placeholder return value
    }
}

File: logout_test.py
# Test for logout functionality

def test_logout():
    # Given the user has successfully logged in
    login()
    
    # When the user clicks the "Logout" button
    click_logout()
    
    # Then the user should be redirected to the login page
    assert current_page() == 'login'
    
    # And a "Logged out successfully" message should be displayed
    assert get_message() == "Logged out successfully"